pudb/run.py

```diff
@@ -5,33 +5,42 @@ def main():
     import sys
 
     from optparse import OptionParser
+    # TODO: update cmd string
     parser = OptionParser(
-            usage="usage: %prog [options] SCRIPT-TO-RUN [SCRIPT-ARGUMENTS]")
+            usage="usage: %prog [options] [SCRIPT-TO-RUN|-m MODULE] [ARGUMENTS]")
 
     parser.add_option("-s", "--steal-output", action="store_true"),
+    parser.add_option("-m", "--module", metavar="MODULE",
+                      help="Debug module or package instead of script"),
     parser.add_option("--pre-run", metavar="COMMAND",
             help="Run command before each program run",
             default="")
     parser.disable_interspersed_args()
     options, args = parser.parse_args()
 
-    if len(args) < 1:
-        parser.print_help()
-        sys.exit(2)
-
-    mainpyfile = args[0]
-
-    from os.path import exists
-    if not exists(mainpyfile):
-        print('Error: %s does not exist' % mainpyfile)
-        sys.exit(1)
-
-    sys.argv = args
-
-    from pudb import runscript
-    runscript(mainpyfile,
-            pre_run=options.pre_run,
-            steal_output=options.steal_output)
+    if options.module:
+        from pudb import runmodule
+        runmodule(options.module,
+                pre_run=options.pre_run,
+                steal_output=options.steal_output)
+    else:
+        if len(args) < 1:
+            parser.print_help()
+            sys.exit(2)
+
+        mainpyfile = args[0]
+
+        from os.path import exists
+        if not exists(mainpyfile):
+            print('Error: %s does not exist' % mainpyfile)
+            sys.exit(1)
+
+        sys.argv = args
+
+        from pudb import runscript
+        runscript(mainpyfile,
+                pre_run=options.pre_run,
+                steal_output=options.steal_output)
```

Comment:
The options processing is a bit redundant. Maybe make an `options_kwargs` above to minimize duplication? If you don't like that, I'm also OK with comments to remind folks to keep the two copies consistent.

Reply:
Sounds good, will do!